src/components/swap/SwapHeader.tsx

```diff
@@ -13,14 +22,136 @@ const StyledSwapHeader = styled.div`
+
+  const [buyFiatClicked, setBuyFiatClicked] = useBuyFiatClicked()
+  const [continueBuyFiatFlowPostRegionCheck, setContinueBuyFiatFlowPostRegionCheck] = useState(false)
+  const [continueBuyFiatFlowPostWalletModal, setContinueBuyFiatFlowPostWalletModal] = useState(false)
+
+  const [checkFiatRegionAvailability, setCheckFiatRegionAvailability] = useState(false)
+  const {
+    available: fiatOnrampAvailable,
+    availabilityChecked: fiatOnrampAvailabilityChecked,
+    error,
+    loading: fiatOnrampAvailabilityLoading,
+  } = useFiatOnrampAvailability(checkFiatRegionAvailability)
+
+  const disableBuyCryptoButton = Boolean(
+    error || (!fiatOnrampAvailable && fiatOnrampAvailabilityChecked) || fiatOnrampAvailabilityLoading
+  )
+
+  const [walletDrawerOpen, toggleWalletDrawer] = useWalletDrawer()
+
+  const handleBuyCrypto = useCallback(() => {
+    if (!fiatOnrampAvailabilityChecked) {
+      setCheckFiatRegionAvailability(true)
+      setContinueBuyFiatFlowPostRegionCheck(true)
+    } else if (fiatOnrampAvailable && !account && !walletDrawerOpen) {
+      toggleWalletDrawer()
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(true)
+    } else if (fiatOnrampAvailable && account) {
+      openFiatOnRampModal()
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    } else if (!fiatOnrampAvailable) {
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    }
+  }, [
+    walletDrawerOpen,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    account,
+    toggleWalletDrawer,
+    openFiatOnRampModal,
+    setBuyFiatClicked,
+  ])
+
+  useEffect(() => {
+    if (continueBuyFiatFlowPostRegionCheck || (account && continueBuyFiatFlowPostWalletModal)) {
+      handleBuyCrypto()
+    }
+  }, [
+    openFiatOnRampModal,
+    account,
+    checkFiatRegionAvailability,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    setBuyFiatClicked,
+    toggleWalletDrawer,
+    handleBuyCrypto,
+    setContinueBuyFiatFlowPostRegionCheck,
+    continueBuyFiatFlowPostWalletModal,
+    continueBuyFiatFlowPostRegionCheck,
+  ])
+
   return (
     <StyledSwapHeader>
       <RowBetween>
         <RowFixed>
-          <ThemedText.DeprecatedBlack fontWeight={500} fontSize={16} style={{ marginRight: '8px' }}>
+          <TextHeader className={subhead} color={theme.textPrimary}>
             <Trans>Swap</Trans>
-          </ThemedText.DeprecatedBlack>
+          </TextHeader>
+          {fiatOnRampButtonEnabled && (
+            <MouseoverTooltipContent
+              wrap={true}
+              content={
+                <div>
+                  <Trans>Crypto purchases are not available in your region. </Trans>
+                  <ExternalLink href={TOKEN_SAFETY_ARTICLE} style={{ paddingLeft: '4px' }}>
```

Comment:
use `gap` on parent instead of defining padding here - in general layout should be parent's responsibility 🤔

Reply:
but also is just the regular sentence space not sufficient here?

Reply:
adding gap in the parent doesn't work in this case, it will treat the link and the sentence as two separate entities and put a space between them. we want it to flow as if one paragraph. the sentence space doesn't show up unfortunately without padding :S I think padding Left here is the best / least code heavy solution.